src/session.rs

```diff
@@ -56,6 +56,34 @@ impl SparkSession {
         DataFrame::new(self, LogicalPlanBuilder::from(range_relation))
     }
 
+    pub fn setCatalog(self: Arc<Self>, catalog: &str) -> DataFrame {
```

Comment:
So for both `setCatalog` and `setDatabase`, these should be implemented on the `spark.catalog` object as `setCurrentCatalog` and `setCurrentDatabase`since we want to mirror the existing Spark API. 

For these actions to take effect on the existing session, the plan has to be submitted to the server via `client.execute_and_fetch` and receive a successful response. Both of these execution plans return nothing from the server.  The code might look like this below

```rust
pub async fn setCurrentCatalog(self, catalog: &str) -> Result<(), SparkError> {
       let cat_type = Some(spark::catalog::CatType::SetCurrentCatalog(
            spark::SetCurrentCatalog { catalog_name: catalog.to_string() },
        ));

        let rel_type = spark::relation::RelType::Catalog(spark::Catalog { cat_type });

        let plan = LogicalPlanBuilder::plan_root(LogicalPlanBuilder::from(rel_type));

        self.spark_session.client().execute_and_fetch(plan).await
}
```